### PR TITLE
backup: global option to tune max no. of backup conf files

### PR DESCRIPTION
**Pull request by pkalever**
Eg:
/> set global backup_files_limit=1000
Parameter backup_files_limit is now '1000'.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

### CODE REVIEW

**Comment by agrover** [comment]
see #96. So this overrides the global-for-all-users pref in the config file with a per-user pref. Is that what we want, or the reverse? (Or something funky like pick the greater of the two?) @tjakob ?

(It's also possible we could improve things in some other way, to make so many backups less needed...if anyone has any ideas.)

**Comment by pkalever** [comment]
> see #96. So this overrides the global-for-all-users pref in the config file with a per-user pref. Is that what we want, or the reverse? (Or something funky like pick the greater of the two?) @tjakob ?

@agrover Make sense to me now. Thanks for correcting me. The Idea about comparing between two and picking the highest looks good to me. I can make the required changes once we agree on it.

> see #96. So this overrides the global-for-all-users pref in the config file with a per-user pref. Is that what we want, or the reverse? (Or something funky like pick the greater of the two?) @tjakob ?

Exactly, I was thinking about doing something like, compare the current and previous saveconfig files and discard pushing save file to backup dir if both files match (No diff)

```
>>> import filecmp 
>>> filecmp.cmp('saveconfig1.json', 'saveconfig2.json')
True
```

**Comment by pkalever** [comment]
> (It's also possible we could improve things in some other way, to make so many backups less needed...if anyone has any ideas.)

Please refer https://github.com/open-iscsi/targetcli-fb/commit/f5054d2120e96bca897c78ecd34b31860f5b826a for the proposed patch

**Comment by agrover** [comment]
Ok, just merged #98. Let me know if you still want/need this global option, since that commit should greatly cut down on redundant backup copies. We also can just bump up default backup copies a little too, if we want.

**Comment by pkalever** [comment]
> Ok, just merged #98. Let me know if you still want/need this global option, since that commit should greatly cut down on redundant backup copies. We also can just bump up default backup copies a little too, if we want.

Thanks @agrover!
I would still prefer to have an option to tune backup file count :-)

**Comment by pkalever** [comment]
@agrover I have updated the PR with your suggestions.

Thanks!

File: targetcli/ui_root.py [hunk 1]
```diff
@@ -99,20 +98,20 @@ def ui_command_saveconfig(self, savefile=default_save_file):
...
-                            with open(universal_prefs_file) as prefs:
-                                backups = [line for line in prefs.read().splitlines() if re.match('^kept_backups\s*=', line)]
-                                kept_backups = int(backups[0].split('=')[1].strip())
-                        except:
-                            kept_backups = default_kept_backups
-
-                        files_to_unlink = list(reversed(backed_files_list))[kept_backups:]
+                        max_backup_files = int(self.shell.prefs['max_backup_files'])
+
+                        with open(universal_prefs_file) as prefs:
```
Comment (tjakob) [inline targetcli/ui_root.py]: I don't think we can get rid of the `try` here, unless we check that the "universal_prefs_file" exists and that we can successfully parse the backups array (which could be empty).

Comment (pkalever) [reply targetcli/ui_root.py]: +1 @tjakob 
I will submit the change soon.

**Comment by pkalever** [comment]
@tjakob @agrover Please find the updated patch[es].

Thanks.

**Comment by agrover** [comment]
👍 from me. @tjakob ?

**Comment by tjakob** [comment]
Looks good to me

**Comment by agrover** [comment]
thanks all!!

**Pull request merged**